### ISSUE TITLE
[eslint-import-resolver-node] node paths doesn't work with yarn berry v3

### ISSUE THREAD

**Issue by rtritto** [import-js/eslint-plugin-import#2188]
# Description
I need to set base path to ESLint to resolve absolute paths.
So I installed `eslint-import-resolver-node` to add, in `eslintrc` file, `node.paths` options in `settings.import/resolver`.
Work only with yarn berry v2 (2.4.2) but not with v3 (3.0.1-rc.2) because I get `import/no-unresolved` error.

# Steps to reproduce (with yarn v3)

1. `yarn init -y`
2. `yarn set version berry`
3. `yarn set version canary`
4. `yarn add -D eslint eslint-config-airbnb eslint-plugin-jsx-a11y eslint-plugin-react eslint-plugin-import eslint-import-resolver-node`
5. Create file `.eslintrc.yml`:
```yml
extends:
  - airbnb
settings:
  import/resolver:
    node:
      paths:
        - '.'
```
6. Create file `src/a.js`:
```js
const b = require('src/b');

const a = () => {
  b();
};

a();

```
7. Create file `src/b.js`:
```js
const b = () => {};

module.exports = b;

```
8. `yarn eslint .`

# Results
## Actual
```
.\import-test\src\a.js
  1:19  error  Unable to resolve path to module 'src/b'  import/no-unresolved

✖ 1 problem (1 error, 0 warnings)
```

## Expected
No errors in log.

# Additional Information
## Versions
eslint: `7.32.0`
eslint-config-airbnb: `18.2.1`
eslint-plugin-jsx-a11y: `6.4.1`
eslint-plugin-react: `7.24.0`
eslint-plugin-import: `2.24.0`
eslint-import-resolver-node: `0.3.5`
yarn: from `3.0.0-rc.1` to `3.0.1-rc.2`
node: `14.17.0`
OS: Windows

**Comment by ljharb** [comment]
Yarn berry doesn’t use node resolution. Like many other things, you should expect this plugin to break when using it.

For this project, someone would have to create an eslint import resolver for PnP, which afaik does not yet exist.

**Comment by rtritto** [comment]
Thanks for answer.
Both yarn berry versions (v2 and v3) use PnP.
It works with last yarn version of v2 (2.4.2) but it breaks only from first yarn version of v3 (v3.0.0-rc.1).
Maybe [commits](https://github.com/yarnpkg/berry/tree/%40yarnpkg/cli%2F3.0.0-rc.1) and [changelog](https://github.com/yarnpkg/berry/blob/%40yarnpkg/cli/3.0.0-rc.1/CHANGELOG.md) can help.

**Comment by ljharb** [comment]
In that case, it seems like a yarn 3 issue - @arcanis, any idea what might be changing here in v3?

**Comment by arcanis** [comment]
I'm not sure why it would change between 2.x and 3.x, but the problem (and its solution) is described here:

https://yarnpkg.com/features/protocols#why-is-the-link-protocol-recommended-over-aliases-for-path-mapping

Short answer, don't use `node.paths`: just add your `src` directory as a dependency via the `link:` protocol (this way all your tools will know how to resolve it, not only eslint), or even better just use your package name in your imports (like you would with any other dependency).

**Comment by rtritto** [comment]
@arcanis thanks for answer.
I removed `settings` in `eslintrc` and added in `package.json`:
```json
...
  "dependencies": {
    "src": "link:./src"
  },
...
```
It not works, I get same error in `eslint` :confused:

**Comment by rtritto** [comment]
I solved using `yarn` command to update `yarn.lock` with `src@link`.
